types/index.d.ts

```diff
@@ -205,7 +205,7 @@ export interface Query {
   search?: string;
   fields?: {[key: string]: string};
   sort?: string;
-  filters?: Filter|AggregatedFilters;
+  filters?: string
```

Comment:
This hurts me a little bit. By integrating TypeScript in our repos, I strove to provide clear guidance and definition on our API to customers. By using this, this makes the Query.filters attribute impossible to use without strong code introspection. 

Do we plan to work on this and change this to a true filter ? (Which already defined and available 😢)

Reply:
> Do we plan to work on this and change this to a true filter ? (Which already defined and available 😢)

The API that is exposed to our customers is low level (customers write routes, middlewares etc), and this type represents the querystring, fresh from express. That's why filters are strings.

At this point it cannot be decoded by us

> By integrating TypeScript in our repos, I strove to provide clear guidance and definition on our API to customers
> Do we plan to work on this and change this to a true filter ? (Which already defined and available 😢)

IMHO, the problem is not the typings, it's the API itself which would gain from being higher level.

I'm a strong proponent for breaking changes in the agents APIs so that customers have a higher level API for next major.

Reply:
Do we have a place where we list stuff to be done in the next major version ? If this is the case, let's log this subject in this place to keep a track on, and validate this PR 👍